docs/build-system.org

```diff
@@ -189,3 +190,563 @@ All your codebase are belong to us.
+3. ~@shuffle~, this function will construct a new vector by selecting the elements of two vectors (a and b) with a mask.
+   If the number of the mask is 0 or upper, it will select the index of the first vector (a).
+   If the number of the mask is -1 or less, it will select the index starting at -1 and decrementing. (read example to be more clear)
+4. ~@select~, this function will select from two vectors (a and b) based on a ~pred~ vector.
+   If ~pred[i]~ is positif, the element will be ~a[i]~ otherwise ~b[i]~
+   
+#+begin_src zig :imports '(std) :main no
+  const stdout = std.io.getStdOut();
+  const size = 4;
+  fn print(comptime T: type, name: []const u8, vector: T) !void {
+      try stdout.writer().print("{s}: ", .{name});
+      for (0..size) |index| {
+          try switch (T) {
+              @Vector(size, bool) => stdout.writer().print("{}, ", .{vector[index]}),
+              @Vector(size, u8) => stdout.writer().print("{c}, ", .{vector[index]}),
+              else => stdout.writer().print("{d}, ", .{vector[index]}),
+          };
+      }
+      try stdout.writer().print("\n", .{});
+  }
+
+  pub fn main() !void {
+      const a: @Vector(size, i32) = @splat(3);
+      try print(@TypeOf(a), "a", a);
+
+      const b = @reduce(.Add, a);
+      try stdout.writer().print("b: {d}\n", .{b});
+
+      const mask: @Vector(size, i32) = .{ 1, -2, -1, 0};
+      const c: @Vector(size, u8) = .{ 't', 't', 'q', 'q'};
+      const d: @Vector(size, u8) = .{ 's', 'e', 't', 'a'};
+      const e = @shuffle(u8, c, d, mask);
+      try print(@TypeOf(e), "e", e);
+
+      const pred: @Vector(size, bool) = .{ true, false, false, true};
+      const f: @Vector(size, u8) = .{ 'a', 'a', 'a', 'a'};
+      const g: @Vector(size, u8) = .{ 'b', 'b', 'b', 'b'};
+      const h = @select(u8, pred, f, g);
+      try print(@TypeOf(h), "h", h);
+  }
+#+end_src
+
+#+RESULTS:
+| a: | 3, | 3, | 3, | 3, |
+| b: | 12 |    |    |    |
+| e: | t, | e, | s, | t, |
+| h: | a, | b, | b, | a, |
+   
+
+*** Leibniz algorithm
+This chapter shows the performance of Zig compared to other language, in particular with C.
+
+The [[https://github.com/niklas-heer/speed-comparison/tree/master][speed-comparaison]] github repository [cite:@NiklasheerSpeedcomparisonRepo] has differents examples code that shows the [[https://en.wikipedia.org/wiki/Leibniz_formula_for_%CF%80][Leibniz algorithm]] [cite:@LeibnizFormula2024] in action.
+This algorithm has puspose to compute the PI number with the following formula:
+
+\begin{equation}
+\frac{\pi}{4} = 1 - \frac{1}{3} + \frac{1}{5} - \frac{1}{7} + \frac{1}{9} - \ldots = \sum_{k = 0}^{\infty} \frac{(-1)^k}{2k + 1}
+\end{equation}
+
+The C code shows the standard way to compute the Leibniz algorithm, at each iteration, PI will improve.
+#+begin_src C :includes <stdio.h> :tangle leibniz.c
+  unsigned rounds = 1000000000;
+  double pi = 1.0;
+
+  rounds += 2u; // do this outside the loop
```

Comment:
maybe you could rewrite it so that readers wont be confused the same way I was

Reply:
It's because PI is initialize at 1 : `pi = 1`.
And to calculate PI you need:

$$ \frac{\pi}{4} = 1 - \frac{1}{3} + \frac{1}{5} - \frac{1}{7} + \frac{1}{9} - \ldots$$

And if `i` start at `0` the calculation will not work.
That's why we add `2`.

Reply:
yes but you added 2 to `rounds` not `pi`

Reply:
Yes because in the loop, we start at 2 until rounds, and to have to good amount of rounds, we add 2.
```c
for (unsigned i=2u; i < rounds; ++i)
```